dhall/src/Dhall/TypeCheck.hs

```diff
@@ -903,7 +903,17 @@ infer typer = loop
             return (VRecord (Dhall.Map.union xRs' xLs'))
 
         RecordCompletion l r -> do
-            loop ctx (Annot (Prefer (Field l "default") r) (Field l "Type"))
+            _L' <- loop ctx l
+
+            case _L' of
+                VRecord xLs' 
+                  | not (Dhall.Map.member "default" xLs')
+                     -> die (InvalidRecordCompletion "default" l)
+                  | not (Dhall.Map.member "Type" xLs')
+                     -> die (InvalidRecordCompletion "Type" l)
+                  | otherwise
+                     -> loop ctx (Annot (Prefer (Field l "default") r) (Field l "Type"))
+                _ -> die (MustCompleteARecord l (quote names _L'))
```

Comment:
IMHO it's `r` that is completed, not `l` – not sure what would be a better error name though!

Reply:
How about 'BadCompletionSchema'?

Reply:
`CompletionSchemaMustBeARecord`?

Reply:
Let's go with that.